File: src/main/kotlin/com/tft/guide/repository/ItemRepository.java
package com.tft.guide.repository;

import com.tft.guide.entity.Item;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;

public interface ItemRepository extends MongoRepository<Item, String>
        , QuerydslPredicateExecutor<Item> {
}

File: src/main/kotlin/com/tft/guide/repository/ItemRepository.kt
package com.tft.guide.repository

import com.tft.guide.entity.Item
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.querydsl.QuerydslPredicateExecutor

interface ItemRepository : MongoRepository<Item, String> {
    fun findAllBySeason(season: String): List<Item>
}